google/resource_sql_user.go

```diff
@@ -111,14 +116,27 @@ func resourceSqlUserRead(d *schema.ResourceData, meta interface{}) error {
 	name := d.Get("name").(string)
 	host := d.Get("host").(string)
 
-	users, err := config.clientSqlAdmin.Users.List(project, instance).Do()
-
-	if err != nil {
-		return handleNotFoundError(err, d, fmt.Sprintf("SQL User %q in instance %q", name, instance))
+	var users *sqladmin.UsersListResponse
+	backoff := 1 * time.Second
+	err = nil
+	for users == nil || err != nil {
+		users, err = config.clientSqlAdmin.Users.List(project, instance).Do()
+
+		if e, ok := err.(*googleapi.Error); ok && (e.Code == 429 || e.Code == 503) {
+			backoff = backoff * 2
```

Comment:
This could also be simplified with `retry`, right? That already retries (with backoff) on 429s and 503s. The only obstacle I'd see is the `users == nil` case- was that one that came up in testing or were you including it just-in-case?

```
err = retry(func() {
  users, err = config.clientSqlAdmin.Users.List(project, instance).Do()
  return err
})
if err != nil {
  return handleNotFoundError(...)
}
```

Reply:
I was using that as an entry condition.  Yours is smarter, changing it.